src/llama-grammar.cpp

```diff
@@ -1127,9 +1225,10 @@ void llama_grammar_accept_impl(struct llama_grammar & grammar, llama_token token
     const auto & code_points = decoded.first;
 
     llama_grammar_stacks stacks_new;
+    llama_grammar_stacks_cache stacks_cache;
```

Comment:
I haven't yet understood the details of the implementation, but just want to confirm that recreating the cache for each token is really what we want? Looks like before moving the cache from global to local state, it was initialized during grammar init and maintained across multiple tokens. Now it is recreated for each token (IIUC).

Reply:
Yes, it is a better option to initialize the cache in grammar_init and return it

Reply:
> Yes, is it better to initialize the cache in grammar_init and return it? 

I don't know yet. I was just looking at the lifetime of the cache and noticed this discrepancy. I will need more time to review the logic of the implementation.

Reply:
I have added `stacks_cache` into `llama_grammar` :)